### ISSUE TITLE
Add More Sorting Options to "Sort By" Dropdown

### ISSUE THREAD

**Issue by Shreyhac** [suryanshsingh2001/MyLibrary#72]
The current "Sort By" dropdown only provides two options: "Newest" and "Relevance." This limits the user’s ability to refine their search based on other important criteria. To improve the user experience and make the sorting feature more comprehensive, I propose adding more sorting options that better cater to user needs, especially for filtering content like books.

Additional Sorting Options:

Oldest: Allow users to sort results starting with the oldest items.
Alphabetical (A-Z): Enable sorting by title or name in ascending alphabetical order.
Alphabetical (Z-A): Allow sorting in descending alphabetical order.
Most Popular: Add an option to sort by popularity or the number of views/purchases.
Rating: Sort by average user ratings (highest to lowest).
Category-Based Sorting (For book-related sections):

Fiction: Filter books by fictional category.
Non-Fiction: Filter books by non-fictional category.
Genre-Specific Filters: Add sorting options like "Fantasy," "Science Fiction," "Romance," etc.
User-Configurable Sorting: Consider adding a feature that lets users customize their default sorting method, providing more control over how they browse content.

Benefits:
Improved User Control: Users can better refine their search, leading to quicker and more relevant results.
Enhanced Usability: Adding more options will make the sorting tool more useful, especially when dealing with larger datasets, such as books or media types.
Broader Filtering: Users looking for specific genres or types of content will have an easier time finding what they need.
<img width="227" alt="Screenshot 2024-10-17 at 01 42 23" src="https://github.com/user-attachments/assets/118aabc7-9e38-4dc8-8f99-f600ff262dd1">

**Comment by Shreyhac** [comment]
please assign me this under hactoberfest

**Comment by suryanshsingh2001** [comment]
@Shreyhac  Absolutely, do make sure these implementations is there in Google Books API as well.
